sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1765,15 +1765,44 @@ def _setup_node(runner: command_runner.SSHCommandRunner) -> int:
                                 stream_logs=False)
                 # Need this `-i` option to make sure `source ~/.bashrc` work
                 cmd = f'/bin/bash -i /tmp/{setup_file} 2>&1'
+                setup_log_path = os.path.join(self.log_dir,
+                                              f'setup-{runner.ip}.log')
                 returncode = runner.run(
                     cmd,
-                    log_path=os.path.join(self.log_dir, 'setup.log'),
+                    log_path=setup_log_path,
                     process_stream=False,
                 )
-                subprocess_utils.handle_returncode(
-                    returncode=returncode,
-                    command=cmd,
-                    error_msg=f'Failed to setup with return code {returncode}')
+
+                def error_message() -> str:
+                    # Use the function to avoid tailing the file in success case
+                    try:
+                        last_10_lines = subprocess.run(
+                            [
+                                'tail', '-n10',
+                                os.path.expanduser(setup_log_path)
+                            ],
+                            stdout=subprocess.PIPE,
+                            check=True).stdout.decode('utf-8')
+                    except subprocess.CalledProcessError:
+                        last_10_lines = None
+
+                    err_msg = (
+                        f'Failed to setup with return code {returncode}. '
+                        f'Check the details in log: {setup_log_path}.')
+                    if last_10_lines:
+                        err_msg += (
+                            f'\n\n{colorama.Fore.RED}'
+                            '****** START Last lines of setup output ******'
+                            f'{colorama.Style.RESET_ALL}\n'
+                            f'{last_10_lines}'
+                            f'\n{colorama.Fore.RED}'
```

Comment:
nit: this `\n` seems redundant since `tail` would have a newline at the end?
```
Failed to setup with return code 1. Check the details in log: ~/sky_logs/sky-2022-08-07-08-33-43-688544/setup-3.235.45.107.log.

****** START Last lines of setup output ******
running setup

******* END Last lines of setup output *******

....

» cat ~/sky_logs/sky-2022-08-07-08-33-43-688544/setup-3.235.45.107.log
running setup
```

Reply:
Good catch! Fixed. Thanks!